spec/function.dd

```diff
@@ -1219,39 +1219,57 @@ $(H2 $(LNAME2 parameters, Function Parameters))
 
 $(H3 $(LNAME2 param-storage, Parameter Storage Classes))
 
-        $(P Parameter storage classes are $(D in), $(D out),
-        $(D ref), $(D lazy), $(D const), $(D immutable), $(D shared),
-        $(D inout) or
-        $(D scope).
+        $(P Parameter storage classes are $(D in), $(D out), $(D ref), $(D lazy), $(D scope) and $(D inout).
+        Parameters can also take the type constructors $(D const), $(D immutable) and $(D shared).)
+
+        $(P $(D in), $(D ref), $(D out) and $(D lazy) are mutually exclusive. The first three are used to
+        denote input, input/output, and output parameters, respectively.
         For example:
         )
 ------
-int foo(in int x, out int y, ref int z, int q);
+int read(in char[] input, ref size_t count, out int errno);
 ------
 
-        $(P x is $(D in), y is $(D out), z is $(D ref), and q is none.
-        )
+        $(P In this example, $(D input) will only be read and no reference to it will be kept around,
+        while $(D count) will be read and written to, and $(D errno) will be set to a value from
+        within the function.
+        This approach gives a semantic meaning to the parameters and allow the compiler to potentially
+        optimize the generated code.)
```

Comment:
allows the compiler to generate code that has more aggressive optimizations based on these assumptions. ?

Reply:
I'm not sure what the difference between those two sentences are, in practice ? I also try to stay away from terms which could have a negative connotation, such as "aggressive".

Reply:
I think I was trying to phrase it in a more direct manner, to make it clear that circumventing the guarantee is not a wrong-code compiler bug.

Reply:
I'm not sure I know which guarantee you refer to ? Maybe "allow the compiler to potentially" => "could allow the compiler to" ?